types/wlr_linux_dmabuf.c

```diff
@@ -212,7 +225,7 @@ static void params_create_common(struct wl_client *client,
 
 		// planes > 0 might be subsampled according to fourcc format
 		if (i == 0 && buffer->attributes.offset[i] +
-				buffer->attributes.stride[i] * height >= size) {
+				buffer->attributes.stride[i] * height > size) {
```

Comment:
Isn't >= correct here since we're starting at 0? In any case this should be in a separate commit.

Reply:
Sooo. I changed this in my previous PR (#1015) and it breaks `weston-simple-dmabuf-drm --import-format=XRGB` (from weston HEAD). Do you think it's a weston bug?

Reply:
Uhh...I missed that change in your previous PR. 
The current code ```>=``` is more strict than just ```>``` or am I missing something? So I wonder how using ```>=``` would make something work that didn't work with just ```>``` ?

Anyways, I'd drop that change.

Reply:
>The current code >= is more strict than just > or am I missing something?

Yeah, the current code is more strict because it makes sense and should work - see all previous checks.

>So I wonder how using >= would make something work that didn't work with just > ?

The current code (`>=`) breaks `weston-simple-dmabuf-drm`. So this change (`>`) unbreaks it.

>Anyways, I'd drop that change.

By "drop", do you mean keep it as strict or make it more lenient just like before #1015?

Reply:
Rereading the code the ```>=``` version is correct. I missed the ```* height``` there. So let's just move the part that turns ```>``` into ```>=``` into a separate commit so we don't sneak in any unrelated changes.

Reply:
I flipped ```>=``` with ```>``` in the above (sorry, to many things going on). I meant to say: let's make it ```>``` in a separate commit since that's correct. stride * height == size is o.k.

Reply:
Okay. Is there any explanation why `stride * height == size` is OK? I assume we'd need to update the check above this one as well, in case of images with height=1.

Reply:
Take the extreme case of height == 1 and width == stride == size. We have pixels [0, width-1] but ```offset + stride == size```. (so maybe the check should read

```
offset + (stride * height) -1 >= size
```

which is

```
offset + (stride * height) > size
```

Reply:
Oh yeah, that does make sense

Reply:
Looks good now. Thanks!